## White Wine Project 

#### How to use this notebook
This notebook contains the analysis of white wine quality. The purpose is to determine what factors influence the quality of white wine.

To use, open this notebook and run the cells from top to bottom and ensure files are in the same folder. The dataset used is sourced from Wine Rating & Price from Kaggle and it is stored in this repository under project.

The files included in this project are White.csv, raw_white_wine.csv,  cleaned_white_wine.csv and results_white_wine.csv.

No external API is required for this notebook.

The libraries needed for this notebook are pandas, numpy, matplotlib and seaborn.

#### Statement of AI
##### A

#### Introduction
This project aims to analyse the various characteristics of white wines to determine which features contribute to a high-quality wine. If it can be determined what factors influence the quality of wine and to what degree, a predictive model could be created to predict the quality of a wine before purchase helping consumers to make more informed decisions.

The dataset was compiled by Anton Budnyak and was originally presented on Vivino.com. It was updated 6 years ago. There are 3,764 different white wines in the data set.

The analysis will focus on the following factors: country, region, winery, price and year. Wine ratings will be used as the primary measure of quality to evaluate what makes a good white wine. 

#### Loading the dataset

In [36]:
import pandas as pd
import numpy as np

df_wine = pd.read_csv("White.csv")

df_wine.head()

,Name,Country,Region,Winery,Rating,NumberOfRatings,Price,Year
0,Vermentino 2017,Italy,Toscana,Famiglia Castellani,3.8,25,5.65,2017
1,Ronco Broilo 2010,Italy,Colli Orientali del Friuli,Conte d'Attimis Maniago,4.3,25,44.90,2010
2,Weisser Schiefer s 2017,Austria,Südburgenland,Weinbau Uwe Schiefer,4.2,25,33.25,2017
3,Chardonnay 2018,Germany,Rheinhessen,Krämer - Straight,3.9,25,8.99,2018
4,Maganza Zibibbo 2018,Italy,Terre Siciliane,Luna Gaia,3.9,25,8.60,2018


#### Exporting the raw dataset

In [37]:
df_wine.to_csv("raw_white_wine.csv", index = False)

# To load this raw dataset uncomment the line below
# df_wine = pd.read_csv("raw_white_wine.csv")

#### Cleaning and transforming the data

##### Inspecting the dataset

In [38]:
df_wine.info()

df_wine.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3764 entries, 0 to 3763
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             3764 non-null   object 
 1   Country          3764 non-null   object 
 2   Region           3764 non-null   object 
 3   Winery           3764 non-null   object 
 4   Rating           3764 non-null   float64
 5   NumberOfRatings  3764 non-null   int64  
 6   Price            3764 non-null   float64
 7   Year             3764 non-null   object 
dtypes: float64(2), int64(1), object(5)
memory usage: 235.4+ KB


,Rating,NumberOfRatings,Price
count,3764.000000,3764.000000,3764.000000
mean,3.817906,187.567216,20.620319
std,0.266827,1071.470441,30.928415
min,2.200000,25.000000,3.740000
25%,3.600000,43.000000,9.260000
50%,3.800000,77.000000,13.150000
75%,4.000000,174.250000,20.865000
max,4.900000,62980.000000,681.370000


From this we can see that there are no missing values, but now we need to check that there are no placeholder values for each column.

##### Checking for duplicates

In [39]:
# Checking each column for the placeholder values 
print(df_wine["Name"].value_counts())
print(df_wine["Country"].value_counts())
print(df_wine["Region"].value_counts())
print(df_wine["Winery"].value_counts())
print(df_wine["Rating"].value_counts())
print(df_wine["NumberOfRatings"].value_counts())
print(df_wine["Price"].value_counts())
print(df_wine["Year"].value_counts())

Name
Chardonnay 2018                                 78
Sauvignon Blanc 2019                            66
Sauvignon Blanc 2018                            51
Chardonnay 2017                                 41
Riesling Trocken 2018                           27
                                                ..
Inzolia - Chardonnay 2018                        1
Gorges 2015                                      1
Juffer Riesling Trocken GG 2018                  1
Zeltinger Sonnenuhr Riesling Auslese*** 2003     1
Vinho Verde Branco N.V.                          1
Name: count, Length: 3040, dtype: int64
Country
Germany           888
Italy             792
France            653
South Africa      296
Spain             256
Austria           243
United States     135
Chile              99
New Zealand        94
Portugal           89
Australia          67
Argentina          56
Romania            12
Switzerland        11
Greece             10
Hungary            10
Slovenia            9
Israel      

From this we can see that the dataset contained no missing values or placeholder entries. All columns contain complete and valid entries. All prices are above 0 and there are no impossible years or negative values where that would not be feasible.

In [40]:
df_wine.duplicated().sum()

np.int64(0)

From this we can see that there are no duplicate rows.

In [41]:
# Changing Year data type from object to float type
df_wine['Year'] = pd.to_numeric(df_wine['Year'], errors='coerce')

# Checking it worked

print(df_wine.dtypes["Year"])


float64


For this project, the number of ratings is not relevant as we are just looking at the overall rating the wine recived so we will drop the number of ratings column.    

In [43]:
df_wine.drop(columns=["NumberOfRatings"], inplace=True)

# Check this dropped the column
df_wine.columns

Index(['Name', 'Country', 'Region', 'Winery', 'Rating', 'Price', 'Year'], dtype='object')

The prices are assumed to be in US dollars as this is standard in Kaggle datasets and no alternative currency is provided. As the prices vary from USD 3.74 to USD 861.37 so to make the analysis comparing cheaper and expensive wines easier to understaned, they will be grouped into bins.

In [48]:
df_wine["PriceCategory"] = pd.cut(
    df_wine["Price"],
    bins = [0, 10, 20, 50, 100, 500],
    labels = ["Budget", "Economy", "Mid", "Premium", "Luxury"]
)

# Checking the code works
df_wine[["Price", "PriceCategory"]].head(6)

,Price,PriceCategory
0,5.65,Budget
1,44.90,Mid
2,33.25,Mid
3,8.99,Budget
4,8.60,Budget
5,15.90,Economy


The data has now been cleaned and transformed and is ready for analysis, to export the cleaned data file see below.

In [49]:
df_wine.to_csv("cleaned_white_wine.csv", index = False)

# To load this cleaned dataset uncomment the line below
# df_wine = pd.read_csv("cleaned_white_wine.csv")